eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ledger.hs

```diff
@@ -244,58 +246,58 @@ ledgerTransition ::
   ) =>
   TransitionRule (someLEDGER era)
 ledgerTransition = do
-  TRC (LedgerEnv slot txIx pp account, LedgerState utxoSt certState, tx) <- judgmentContext
+  TRC (LedgerEnv slot txIx pp account, LedgerState utxoState certState, tx) <- judgmentContext
   let txBody = tx ^. bodyTxL
+      dstate = certDState certState
+      genCerts = dsGenDelegs dstate
 
-  certState' <-
+  (utxoState', certState') <-
     if tx ^. isValidTxL == IsValid True
-      then
-        trans @(EraRule "CERTS" era) $
-          TRC
-            ( DelegsEnv slot txIx pp tx account
-            , certState
-            , StrictSeq.fromStrict $ txBody ^. certsTxBodyL
-            )
-      else pure certState
-
-  let wdrlAddrs = Map.keysSet . unWithdrawals $ tx ^. bodyTxL . withdrawalsTxBodyL
-  let wdrlCreds = Set.map getRwdCred wdrlAddrs
-  let dUnified = dsUnified $ certDState certState'
-      delegatedAddrs = DRepUView dUnified
-
-  -- TODO enable this check once delegation is fully implemented in cardano-api
-  when False $ do
-    all (`UMap.member` delegatedAddrs) wdrlCreds
-      ?! ConwayWdrlNotDelegatedToDRep (wdrlCreds Set.\\ Map.keysSet (dRepMap dUnified))
+      then do
+        certState' <-
+          trans @(EraRule "CERTS" era) $
+            TRC
+              ( DelegsEnv slot txIx pp tx account
+              , certState
+              , StrictSeq.fromStrict $ txBody ^. certsTxBodyL
+              )
+        let wdrlAddrs = Map.keysSet . unWithdrawals $ tx ^. bodyTxL . withdrawalsTxBodyL
+        let wdrlCreds = Set.map getRwdCred wdrlAddrs
+        let dUnified = dsUnified $ certDState certState'
```

Comment:
```suggestion
        let wdrlAddrs = Map.keysSet . unWithdrawals $ tx ^. bodyTxL . withdrawalsTxBodyL
            wdrlCreds = Set.map getRwdCred wdrlAddrs
            dUnified = dsUnified $ certDState certState'
```

Please ignore if unnecessary.

Reply:
I like this change, but I don't want to rerun the CI again :grin: 

I'll apply it some other PR :wink: